analyses/chromosomal-instability/util/chr-break-calculate.R

```diff
@@ -185,8 +193,36 @@ break_density <- function(breaks_df = NULL,
   bins@elementMetadata@listData$total_counts <- bin_counts
 
   # Calculate and store density
-  bins@elementMetadata@listData$density <- bin_counts / window_size
-
+  bins@elementMetadata@listData$density <- bin_counts / bins@ranges@width
+
+  # If specified, make unsurveyed into a GenomicRanges
+  if (!is.null(unsurveyed_bed)) {
+    # Make GRanges for uncallable ranges
+    unsurveyed_ranges <- GenomicRanges::GRanges(
+      seqnames = unsurveyed_bed$chrom,
+      ranges = IRanges::IRanges(
+        start = unsurveyed_bed$start,
+        end = unsurveyed_bed$end
+      ),
+      total_counts = NA
+    )
+
+    # Find the NA region(s) for each bin without combining close regions
+    na_regions <- GenomicRanges::pintersect(IRanges::findOverlapPairs(bins, unsurveyed_ranges))
+
+    # Find overlap between na_regions and bin
+    na_overlaps <- GenomicRanges::findOverlaps(bins, na_regions)
+
+    # Calculate the percent overlap, adding up the na regions within a bin
+    pct_overlap <- tapply(na_regions@ranges@width, na_overlaps@from, sum) / bins[unique(na_overlaps@from)]@ranges@width
```

Comment:
```suggestion
    pct_overlap <- tapply(na_regions@ranges@width, na_overlaps@from, sum) / 
                     bins[unique(na_overlaps@from)]@ranges@width
```

Reply:
Can you explain what each part of this calculation is doing? I am confused, not being intimately familiar with GenomicRanges

Reply:
I split this up into more steps with more comments. Let me know where else I can make this PR less confusing.